src/main/java/com/google/sps/servlets/UserServlet.java

```diff
@@ -56,6 +59,39 @@ public void init() {
     db = new FirestoreDB();
   }
 
+  /** Currently lets the front end know if a recipe is saved by a user or not. */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String recipeID = request.getParameter("recipeID");
```

Comment:
Use
```
String[] recipeIds = request.getParameterValues("recipeID");
```
instead. This allows for a single request to ask the save status of multiple recipes, instead of sending multiple requests. This will save a bunch of time when loading information on the frontend.

Reply:
I don't think we'd ever need to get the save status of a bunch of recipes at once--this is specifically for recipe-display page which will apply to one recipe at a time. Also, the loading of recipe-display seems to take a negligible amt of time, pul & check it out

Reply:
I have a save icon for each recipe on the feed page, so this applies to that. We don't want to have 20 different requests just for saved recipes on the feed page.

Reply:
Ah ok so i assume this would just take recipe ids in order from the feed page and order the query string/return array that way? Do you want me to implement the front end too

Reply:
(idk if the reactjs will let you have a save icon on recipe display as well, but that's what save/unsave looks like for now)

Reply:
Yeah, its not super hard to do because i getUserProperty is overloaded with an implementation for an array of recipeIds.
Don't worry about frontend implementation for now, I'll hopefully merge the react branch in soon.